google-cloud-spanner/src/test/java/com/google/cloud/spanner/InstanceAdminClientImplTest.java

```diff
@@ -364,22 +398,119 @@ public void testUpdateInstanceProcessingUnits() throws Exception {
                 "updateInstance",
                 getInstanceProtoWithProcessingUnits(),
                 UpdateInstanceMetadata.getDefaultInstance());
-    when(rpc.updateInstance(instance, FieldMask.newBuilder().addPaths("processing_units").build()))
+    when(rpc.updateInstance(
+            instance,
+            FieldMask.newBuilder()
+                .addPaths("autoscaling_config")
+                .addPaths("processing_units")
+                .build()))
+        .thenReturn(rawOperationFuture);
+    InstanceInfo instanceInfo =
+        InstanceInfo.newBuilder(InstanceId.of(INSTANCE_NAME))
+            .setInstanceConfigId(InstanceConfigId.of(CONFIG_NAME))
+            .setProcessingUnits(10)
+            .build();
+    OperationFuture<Instance, UpdateInstanceMetadata> operationWithFieldMask =
+        client.updateInstance(
+            instanceInfo,
+            InstanceInfo.InstanceField.AUTOSCALING_CONFIG,
+            InstanceInfo.InstanceField.PROCESSING_UNITS);
+    assertTrue(operationWithFieldMask.isDone());
+    assertEquals(INSTANCE_NAME, operationWithFieldMask.get().getId().getName());
+
+    when(rpc.updateInstance(
```

Comment:
* Refactor this to a new test. A single test should have a single outcome. Also, let's not modify the behaviour of any existing tests.
* In general I see a lot of modifications to existing tests where we are adding a new `when().then()` mocking and then doing an additional `update`. Can you describe why we are doing it?

Reply:
```Refactor this to a new test. A single test should have a single outcome. Also, let's not modify the behaviour of any existing tests. ```
Yeah sorry this was for my local test -- reverted.

``` In general I see a lot of modifications to existing tests where we are adding a new when().then() mocking and then doing an additional update. Can you describe why we are doing it? ```
This pattern seems to be there before: https://screenshot.googleplex.com/VzKP68q9X5CZSjq ? I was following the same pattern.